src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -355,6 +355,9 @@ public void onFailure(Exception e) {
                 listener.onFailure(new ElasticsearchSecurityException(err, RestStatus.FORBIDDEN));
                 return;
             }
+        } catch (ElasticsearchException e) {
+            log.debug("Failed to apply filter. Task id: {} ({}). Action: {}", task.getId(), task.getDescription(), action, e);
```

Comment:
May `task` be `null`?

Reply:
`task` is not going to be `null` here. Otherwise, other usages of `task` prior to this line in this method will simply complain null pointer exception.

Reply:
Where previously are we accessing a member of task? null check should be handled in the functions it is passed to

Reply:
A lot of usages of task in this function, such as [here](https://github.com/opendistro-for-elasticsearch/security/blob/f382a820ad8512c878a1eaa85bf604e2d4aa4b7d/src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java#L319)

Reply:
We are not accessing a member of the object there. Elasticsearch proceed function will handle the null task. Here we are directly accessing a class member `task.getId()`, `task.getDescription()`

Reply:
Yes, this is indeed the only place where the members of task are accesses. I've added the null check.